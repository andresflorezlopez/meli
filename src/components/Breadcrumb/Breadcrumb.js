import React from "react";
import PropTypes from "prop-types";

import Typography from "@components/Typography/Typography";
import { StyledBreadcrumb, StyledCategory } from "./Breadcrumb.styled";

const Breadcrumb = ({ categories }) => {
  const renderCategories = () =>
    categories.map(category => (
      <StyledCategory>
        <Typography type="caption"> {category} </Typography>
      </StyledCategory>
    ));
  return <StyledBreadcrumb>{renderCategories()}</StyledBreadcrumb>;
};

Breadcrumb.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Breadcrumb;
