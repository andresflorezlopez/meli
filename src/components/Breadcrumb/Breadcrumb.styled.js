import styled from "styled-components";

export const StyledBreadcrumb = styled.div`
  display: flex;
  flex-direction: row;
`;

export const StyledCategory = styled.div`
  cursor: pointer;
  color: ${({ theme }) => theme.colors.font.light};
  position: relative;
  padding-right: ${({ theme }) => theme.spacing(3)};
  &:hover {
    color: ${({ theme }) => theme.colors.font.normal};
  }
  &:after {
    position: absolute;
    content: "â€º";
    top: -4px;
    font-size: 20px;
    right: ${({ theme }) => theme.spacing(1)};
  }
  &:last-child {
    &:after {
      content: "";
    }
  }
`;
