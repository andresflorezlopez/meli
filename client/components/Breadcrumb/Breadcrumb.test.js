import React from "react";
import Breadcrumb from "./Breadcrumb";
import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("Breadcrumb", () => {
  test("should render", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <Breadcrumb
          categories={["Video juegos", "Nintendo", "Nintendo switch"]}
        />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
