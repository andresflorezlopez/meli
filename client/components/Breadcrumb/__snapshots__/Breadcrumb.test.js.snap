// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render 1`] = `
<div
  className="qg2tkp-0 hIrIhx"
>
  <div
    className="qg2tkp-1 eOYxwL"
  >
    <p
      className="sc-1hvrgbn-5 frCBin"
      color="light"
      fontWeight="normal"
      type="caption"
    >
      Video juegos
    </p>
  </div>
  <div
    className="qg2tkp-1 eOYxwL"
  >
    <p
      className="sc-1hvrgbn-5 frCBin"
      color="light"
      fontWeight="normal"
      type="caption"
    >
      Nintendo
    </p>
  </div>
  <div
    className="qg2tkp-1 eOYxwL"
  >
    <p
      className="sc-1hvrgbn-5 frCBin"
      color="light"
      fontWeight="normal"
      type="caption"
    >
      Nintendo switch
    </p>
  </div>
</div>
`;
